### PR TITLE
Improve page, component and layout generators with idiomatic TypeScript support

### PR DESCRIPTION
**Pull request by zakmandhro**
- All generators now create pure TypeScript functions - no React.FunctionComponent anti-pattern (see [post](https://fettblog.eu/typescript-react-why-i-dont-use-react-fc/))
- All 3 generators now use tsx templates, instead of a mix of .js and .tsx.
- Proper prop-types added to Layout and Page generators
- Page generator supports taking strongly typed prop type from command line option
- All tests and snapshots updated (with a few small but obvious fixes in tests)

### CODE REVIEW

File: packages/cli/src/commands/generate/component/templates/component.tsx.template [hunk 1]
```diff
@@ -1,4 +1,4 @@
-const ${pascalName}: React.FunctionComponent = () => {
+const ${pascalName} = () => {
```
Comment (peterp) [inline packages/cli/src/commands/generate/component/templates/component.tsx.template:1]: 👏

**Comment by peterp** [comment]
This is great, thanks for this!

**Review by peterp — APPROVED** [review]
LGTM

**Comment by dac09** [comment]
Very solid improvement here @zakmandhro, will merge shortly after release!

**Review by dac09 — APPROVED** [review]

**Comment by zakmandhro** [comment]
Thanks @dac09 and @peterp! Glad you like the PR. Looking forward to more contributions.

Does this mean I get to see my name on the coveted All Contributors list? 😄

**Pull request merged**